src/libsyncengine/syncpal/syncpal.cpp

```diff
@@ -788,6 +788,23 @@ void SyncPal::directDownloadCallback(UniqueId jobId) {
         return;
     }
 
+    if (directDownloadJobsMapIt->second->getStatusCode() == Poco::Net::HTTPResponse::HTTP_NOT_FOUND) {
+        Error error(syncDbId()
```

Comment:
Just a thought for later: we may define derived error classes with suggestive names and simpler constructors. In this specific case, we would ideally pass only three "dynamic parameters" to the constructor let all the remaining parameters be initialized in the derived class constructor.

Reply:
Yes I agree. The errors will also need refactoring to simplify its use.
Why not a very simple constructor and a few setters to specialize the error?

Reply:
Done, but this is only a minor changes in the error management system, we should deeply refactor it later